pkg/dedup/merge_iter.go

```diff
@@ -92,18 +92,23 @@ func (m *mergedSeriesIterator) At() (t int64, v float64) {
 	return m.lastIter.At()
 }
 
-func (it *mergedSeriesIterator) AtHistogram(h *histogram.Histogram) (int64, *histogram.Histogram) {
-	return it.lastIter.AtHistogram(h)
+func (m *mergedSeriesIterator) AtHistogram(h *histogram.Histogram) (int64, *histogram.Histogram) {
+	return m.lastIter.AtHistogram(h)
 }
 
-func (it *mergedSeriesIterator) AtFloatHistogram(fh *histogram.FloatHistogram) (int64, *histogram.FloatHistogram) {
-	return it.lastIter.AtFloatHistogram(fh)
+func (m *mergedSeriesIterator) AtFloatHistogram(fh *histogram.FloatHistogram) (int64, *histogram.FloatHistogram) {
+	return m.lastIter.AtFloatHistogram(fh)
```

Comment:
Should we do the same to those two methods?

Reply:
see comments below, we do want to panic if At() is called when no lastIter(). This bug is only valid for `Err()` because it can be called even lastIter was not getting a single value.

```
// Err All At() funcs should panic if called after Next() or Seek() return ValNone.
// Only Err() should return nil even after Next() or Seek() return ValNone.
```